internal/cli/ondemand_runner_apply.go

```diff
@@ -0,0 +1,260 @@
+	od.Default = c.flagDefault
+
+	for _, kv := range c.flagEnvVars {
+		idx := strings.IndexByte(kv, '=')
+		if idx != -1 {
+			od.EnvironmentVariables[kv[:idx]] = kv[idx+1:]
+		}
+	}
+
+	od.PluginType = c.flagPluginType
+
+	// Upsert
+	_, err := c.project.Client().UpsertOndemandRunner(ctx, &pb.UpsertOndemandRunnerRequest{
+		OndemandRunner: od,
+	})
+	if err != nil {
+		c.ui.Output(
+			"Error upserting ondemand runner: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	if updated {
+		s.Update("Ondemand Runner updated")
+	} else {
+		s.Update("Ondemand Runner created")
+	}
+	s.Done()
+
+	return 0
+}
+
+func (c *OndemandRunnerApplyCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(sets *flag.Sets) {
+		f := sets.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:    "id",
+			Target:  &c.flagId,
+			Default: "",
+			Usage:   "The id an existing ondemand runner to update.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "oci-url",
+			Target:  &c.flagOCIUrl,
+			Default: "hashicorp/waypoint:stable",
+			Usage:   "The url for the OCI image to launch for the ondemand runner.",
+		})
+
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   "env-vars",
+			Target: &c.flagEnvVars,
+			Usage: "Environment variable to expose to the ondemand runner. Typically used to " +
+				"introduce configuration for the plugins that the runner will execute.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "plugin-type",
+			Target:  &c.flagPluginType,
+			Default: "",
+			Usage:   "The type of the plugin to launch for the ondemand runner, such as aws-ecs, kubernetes, etc.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "project-config",
+			Target:  &c.flagPluginConfig,
+			Default: "",
+			Usage: "Path to a an hcl file that contains the configuration for the plugin." +
+				"This is only necessary when the plugin's defaults need to be adjusted for " +
+				"the environment the plugin will launch the ondemand runner.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "default",
+			Target:  &c.flagDefault,
+			Default: false,
+			Usage: "Indicates that this ondemand runner should be used by any project that doesn't " +
+				"otherwise specify it's own ondemand runner.",
+		})
+	})
+}
+
+func (c *OndemandRunnerApplyCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *OndemandRunnerApplyCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *OndemandRunnerApplyCommand) Synopsis() string {
+	return "Create or update an ondemand runner."
+}
+
+func (c *OndemandRunnerApplyCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint ondemand-runner apply [OPTIONS]
+
+  Create or update an ondemand runner.
+
+  This will create a new ondemand runner with the given options. If
+  a ondemand runner with the same id already exists, this will update the
+  existing runner using the fields that are set.
```

Comment:
```suggestion
  Create or update an ondemand runner configuration.

  This will register a new ondemand runner config with the given options. If
  a ondemand runner config with the same id already exists, this will update the
  existing runner config using the fields that are set.
  
  Waypoint will use an ondemand runner configuration to spawn containers for
  various kinds of work as needed on the platform requested during any given
  lifecycle operation.
```